File: backend/app/api/v1/skill_passport.py
from typing import Annotated

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.dependencies import require_candidate
from app.db.session import get_db
from app.models.user import User
from app.schemas.skill_passport import SkillPassportResponse
from app.services.candidate import get_candidate_profile
from app.services.skill_passport import build_passport, empty_passport

router = APIRouter(prefix="/candidate/skill-passport", tags=["skill-passport"])


@router.get("", response_model=SkillPassportResponse)
def get_skill_passport(
    current_user: Annotated[User, Depends(require_candidate)],
    session: Annotated[Session, Depends(get_db)],
) -> SkillPassportResponse:
    profile = get_candidate_profile(session, current_user.id)
    if profile is None:
        return empty_passport()
    return build_passport(session, profile.id)

File: frontend/lib/api/skill-passport.ts
import { apiRequest } from "@/lib/api/client";
import type { SkillPassportResponse } from "@/lib/api/types/skill-passport";

export function getSkillPassport(): Promise<SkillPassportResponse> {
  return apiRequest<SkillPassportResponse>("/candidate/skill-passport");
}
